### DISCUSSION TITLE
Getting an error while spin develpment server in ubuntu 20.04?

### DISCUSSION THREAD

**Discussion by priyanshu-kun** [kimlimjustin/xplorer#189] [General]
As I has say in the title when I run the development server in ubuntu 20.04 I've got 

```
Uncaught TypeError: Cannot read properties of undefined (reading '__currentWindow')
    at Object.s [as getCurrent] (window-cf408aca.cjs:1)
    at eval (window.ts:6)
    at Object../src/Api/window.ts (index.js:194)
    at __webpack_require__ (index.js:4811)
    at eval (windowManager.ts:28)
    at Object../src/Components/Layout/windowManager.ts (index.js:711)
    at __webpack_require__ (index.js:4811)
    at eval (index.ts:7)
    at Object../src/index.ts (index.js:975)
    at __webpack_require__ (index.js:4811)

```
this following error can anybody help me?

And seems like my development application breaks completely!

**Comment by lg-nineoneninethreenine** [comment]
Well, are you using a local machine or Gitpod? If you are using a local machine, you need Rust, Tauri dependencies, nvm-sh(for node version management). Gitpod update is WIP, I'm having a bad time with it.
Since Gitpod uses Ubuntu, if it works there, I'll do tell here!

**Comment by kimlimjustin** [comment]
Some questions for you:
1. Have you installed all dependencies?
2. If it does, is the error occur on the terminal, or do you open the local dev server in the browser? (It's more likely it's the first case tho)

**Reply by priyanshu-kun** [reply]
I didn't face any error in the terminal but when I run the local dev server and open `http://127.0.0.1:8080/` following URL in the browser there I got the error that above I mention in the console.

![This is the application in browser](https://imgur.com/a/BZrVbPA)

**Reply by kimlimjustin** [reply]
Isn't there any window of Xplorer pop up?

Also, you're not supposed to open that URL in the browser cuz there are many communication functions that are only allowed to run on the development window, and the purpose of the URL is actually just for providing (tricking per se) a hot-reload function for the dev process.

**Reply by priyanshu-kun** [reply]
uhh! I don't think there is any window of Xplorer pop after the spin dev server. when I try to debug I've got this new error 
```
[3] Caused by:
[3]   process didn't exit successfully: `/home/User/Desktop/projects/Opensource Projects/Xplorer/src-tauri/target/debug/build/soup-sys-ae1cb7c55274b755/build-script-build` (exit status: 1)
[3]   --- stdout

```

**Reply by kimlimjustin** [reply]
Could you please run the following command and post the result:
```bash
yarn tauri info
```
This might be because of some incompatible dependencies, you need a nightly version of Rust to get it to works.
Also, there's a core bug I've just fixed minutes ago, could you please pull it before trying?
